src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,325 @@
+        using VectorNormsEstimator       = Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>;
+        // ----------------------------------------------------------------------
+
+        VectorNormsAnnotationData const & data(VectorNormsEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, Components::VectorNormsEstimatorName));
+
+        assert(data.Type == NormT);
+        return typename BaseType::TransformerUniquePtr(new NormalizeTransformer<InputT>(data.Norms));
+    }
+};
+
+} // namespace Details
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         NormalizeEstimator
+///  \brief         Chains up VectorNormsEstimator and NormalizeTransformerImpl
+///                 to normalize input vector data so that it's norm becoming 1
+///
+template <
+    typename InputT,
+    NormType NormT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class NormalizeEstimator :
+    public Components::PipelineExecutionEstimatorImpl<
+        Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>,
+        Details::NormalizeEstimatorImpl<InputT, NormT, MaxNumTrainingItemsV>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::PipelineExecutionEstimatorImpl<
+            Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>,
+            Details::NormalizeEstimatorImpl<InputT, NormT, MaxNumTrainingItemsV>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    NormalizeEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~NormalizeEstimator(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(NormalizeEstimator);
+};
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  NormalizeTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(std::vector<std::double_t> norms) :
+    _norms(std::move(norms)),
+    _row(0) {
+        for (auto it = norms.begin(); it != norms.end(); ++it) {

```

Comment:
>norms [](start = 23, length = 5)

norms will be invalid here, as it has been moved to _norms. You will want to operate only on _norms.